Framework/PythonInterface/mantid/utils/absorptioncorrutils.py

```diff
@@ -27,12 +33,146 @@ def _getBasename(filename):
+    :param cache_dir: location to store the cached absorption correction
+    
+    return fileName(full path), sha1
+    """
+
+    # fix up the workspace name
+    prefix = wkspname.replace('__', '')
+
+    # parse algorithm used for absorption calculation
+    # NOTE: the actual algorithm used depends on selected abs_method, therefore
+    #       we are embedding the algorithm name into the SHA1 so that we can
+    #       differentiate them for caching purpose
+    alg_name = {
+        "SampleOnly": "AbsorptionCorrection",
+        "SampleAndContainer": "AbsorptionCorrection",
+        "FullPaalmanPings": "PaalmanPingsAbsorptionCorrection",
+    }[abs_method]
+
+    # key property to generate the HASH
+    ws = mtd[wkspname]
+    # NOTE:
+    #  - The query for height is tied to a beamline, which is not a good design as it
+    #    will break for other beamlines
+    property_string = [
+        f"{key}={val}" for key, val in {
+            'wavelength_min': ws.readX(0).min(),
+            'wavelength_max': ws.readX(0).max(),
+            "num_wl_bins": len(ws.readX(0)) - 1,
+            "sample_formula": ws.run()['SampleFormula'].lastValue().strip(),
+            "mass_density": ws.run()['SampleDensity'].lastValue(),
+            "height_unit": ws.run()['BL11A:CS:ITEMS:HeightInContainerUnits'].lastValue(),
+            "height": ws.run()['BL11A:CS:ITEMS:HeightInContainer'].lastValue(),
+            "sample_container": ws.run()['SampleContainer'].lastValue().replace(" ", ""),
+            "algorithm_used": alg_name,
+        }.items()
+    ]
+
+    cache_path, signature = mantid.simpleapi.CreateCacheFilename(
+        Prefix=prefix,
+        OtherProperties=property_string,
+        CacheDir=cache_dir,
+    )
+
+    return cache_path, signature
+
+
+def _getCachedData(absName, abs_method, sha1, cache_file_name):
+    """
+    Check both memory and disk to locate cache data.  Returns ("", "")
+    if no cache can be found.
+
+    :params absName: absorption workspace name base
+    :params abs_method: absorption calculation method
+    :params sha1: SHA1 that identify cached workspace
+    :params cache_file_name: cache file name to search
+
+    return cached_absorption_correction_workspace_name_sample,
+           cached_absorption_correction_workspace_name_container
+    """
+    wsn_as = ""  # absorption workspace sample
+    wsn_ac = ""  # absorption workspace container
+    found_as = False
+    found_ac = False
+
+    # step_0: depending on the abs_method, suffix will be different
+    if abs_method == "SampleOnly":
+        wsn_as = f"{absName}_ass"
+        found_ac = True
+    elif abs_method == "SampleAndContainer":
+        wsn_as = f"{absName}_ass"
+        wsn_ac = f"{absName}_acc"
+    elif abs_method == "FullPaalmanPings":
+        wsn_as = f"{absName}_assc"
+        wsn_ac = f"{absName}_ac"
+    else:
+        raise RuntimeWarning("Unrecognized absorption correction method '{}'".format(abs_method))
```

Comment:
This should be a `RuntimeError` rather than [`RuntimeWarning`](https://docs.python.org/3/library/exceptions.html#RuntimeWarning) since you aren't pointing out "dubious behavior".

Reply:
elevated the exception level to ValueError